src/NuGet.Core/NuGet.Protocol/Utility/ProtocolDiagnosticsStream.cs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.Protocol.Utility
+{
+    internal sealed class ProtocolDiagnosticsStream : Stream
+    {
+        private readonly Stream _baseStream;
+        private ProtocolDiagnosticInProgressEvent _inProgressEvent;
+        private readonly Stopwatch _stopwatch;
+        private long _bytes;
+        private readonly Action<ProtocolDiagnosticEvent> _event;
+
+        internal ProtocolDiagnosticsStream(Stream baseStream, ProtocolDiagnosticInProgressEvent inProgressEvent, Stopwatch stopwatch, Action<ProtocolDiagnosticEvent> @event)
```

Comment:
please don't name things event :(

Reply:
I'll rename it in my next PR. I don't want to wait for another CI build just to change this :(

Reply:
🙈 🙉 🙊